src/java.base/share/classes/java/lang/String.java

```diff
@@ -1874,6 +1874,9 @@ private boolean nonSyncContentEquals(AbstractStringBuilder sb) {
         byte v2[] = sb.getValue();
         byte coder = coder();
         if (coder == sb.getCoder()) {
+            if (v1 == v2) {
```

Comment:
How is this possible? Array inside a StringBuilder is modifiable, while array inside String is immutable.

Reply:
You are right. Thank you.